zeebe/stream-platform/src/main/java/io/camunda/zeebe/stream/impl/BufferedProcessingResultBuilder.java

```diff
@@ -115,6 +119,11 @@ public boolean canWriteEventOfLength(final int eventLength) {
     return mutableRecordBatch.canAppendRecordOfLength(eventLength);
   }
 
+  public ProcessingResultBuilder withOperationReference(final long operationReference) {
+    this.operationReference = operationReference;
+    return this;
+  }
```

Comment:
🔧 The value is only set once and is then left unchanged, which means we could make the field final ([preferred](https://github.com/camunda/camunda/wiki/Code-Style#favor-final-modifier)). You could consider overloading the constructor if it must remain optional.